azurelinuxagent/distro/default/osutil.py

```diff
@@ -261,6 +261,9 @@ def set_selinux_context(self, path, con):
         Returns exit result.
         """
         if self.is_selinux_system():
+            if not os.path.exists(path):
```

Comment:
I feel like this checks should not be inside the `set_selinux_context`/`chcon`/`chown` methods. Why don't we check the file before they get called? It does not make much sense to call multiple operations on a file that does not exist, plus this patch adds non-obvious logic to method names like chown/chmod.